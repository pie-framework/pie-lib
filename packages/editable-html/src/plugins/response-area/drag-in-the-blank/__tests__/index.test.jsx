import React from 'react';
import { Data } from 'slate';
import { shallow } from 'enzyme';
import MockChange, { MockDocument } from '../../__tests__/mock-change';
import DragInTheBlank, { onValueChange, onRemoveResponse } from '../index';
describe('choice', () => {
  describe('snapshot', () => {
    let attributes = { key: '0' },
      n = { key: '0' },
      nodeProps = { children: 'A' },
      opts = { options: { duplicates: false } };

    const mkWrapper = (Component, props) => {
      return shallow(<Component {...props} />);
    };

    it('renders correctly when not in a table and duplicates are off', () => {
      const w = mkWrapper(DragInTheBlank, {
        attributes,
        data: { inTable: false },
        n,
        nodeProps,
        opts
      });

      expect(w).toMatchSnapshot();
    });

    it('renders correctly when in a table and duplicates are off', () => {
      const w = mkWrapper(DragInTheBlank, {
        attributes,
        data: { inTable: true },
        n,
        nodeProps,
        opts
      });

      expect(w).toMatchSnapshot();
    });

    it('renders correctly when in a table and duplicates are on', () => {
      const w = mkWrapper(DragInTheBlank, {
        attributes,
        data: { inTable: true },
        n,
        nodeProps,
        opts: { options: { duplicates: true } }
      });

      expect(w).toMatchSnapshot();
    });
  });
  describe('handlers', () => {
    let nodeProps,
      n,
      document = new MockDocument({
        docFindDescendant: jest.fn().mockReturnValue({
          key: '0',
          data: Data.create({ index: '0' })
        })
      }),
      change,
      onChange,
      onEditingDone;

    beforeEach(() => {
      n = { key: '0', data: Data.create({ index: '0' }) };

      onChange = jest.fn();
      onEditingDone = jest.fn();

      change = new MockChange();

      nodeProps = {
        children: 'A',
        editor: {
          props: { onChange, onEditingDone },
          value: {
            document,
            change: () => change
          }
        }
      };
    });

    describe('onValueChange', () => {
      it('should change value', () => {
        onValueChange(nodeProps, n, { value: 'B' });
        expect(change.setNodeByKey).toBeCalledWith('0', {
          data: {
            index: '0',
            value: 'B'
          }
        });

        expect(onChange).toBeCalledWith(change, expect.anything());
        expect(onChange.mock.calls[0][1]).toBeInstanceOf(Function);

        onChange.mock.calls[0][1]();

        expect(onEditingDone).toBeCalled();
      });
    });
    describe('onRemoveResponse', () => {
      it('should remove response', () => {
        onRemoveResponse(nodeProps, n, { value: 'B' });
        expect(change.setNodeByKey).toBeCalledWith('0', {
          data: { index: '0' }
        });
        expect(onChange).toBeCalledWith(change, expect.anything());
        expect(onChange.mock.calls[0][1]).toBeInstanceOf(Function);
        onChange.mock.calls[0][1]();
        expect(onEditingDone).toBeCalled();
      });
    });
  });
});
