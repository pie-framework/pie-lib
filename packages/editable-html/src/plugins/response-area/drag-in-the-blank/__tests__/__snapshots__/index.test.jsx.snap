// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`choice snapshot renders correctly when in a table and duplicates are off 1`] = `
<span
  key="0"
  style={
    Object {
      "cursor": "pointer",
      "display": "inline-flex",
      "margin": "10px",
      "minHeight": "50px",
      "minWidth": "178px",
      "position": "relative",
    }
  }
>
  <DragSource(DropTarget(WithStyles(Component)))
    dragKey="0"
    duplicates={false}
    n={
      Object {
        "key": "0",
      }
    }
    onChange={[Function]}
    removeResponse={[Function]}
    targetId="0"
    value={
      Object {
        "inTable": true,
      }
    }
  >
    A
  </DragSource(DropTarget(WithStyles(Component)))>
</span>
`;

exports[`choice snapshot renders correctly when in a table and duplicates are on 1`] = `
<span
  key="0"
  style={
    Object {
      "cursor": "pointer",
      "display": "inline-flex",
      "margin": "10px",
      "minHeight": "50px",
      "minWidth": "178px",
      "position": "relative",
    }
  }
>
  <DragSource(DropTarget(WithStyles(Component)))
    dragKey="0"
    duplicates={true}
    n={
      Object {
        "key": "0",
      }
    }
    onChange={[Function]}
    removeResponse={[Function]}
    targetId="0"
    value={
      Object {
        "inTable": true,
      }
    }
  >
    A
  </DragSource(DropTarget(WithStyles(Component)))>
</span>
`;

exports[`choice snapshot renders correctly when not in a table and duplicates are off 1`] = `
<span
  key="0"
  style={
    Object {
      "cursor": "pointer",
      "display": "inline-flex",
      "margin": "0 10px",
      "minHeight": "50px",
      "minWidth": "178px",
      "position": "relative",
    }
  }
>
  <DragSource(DropTarget(WithStyles(Component)))
    dragKey="0"
    duplicates={false}
    n={
      Object {
        "key": "0",
      }
    }
    onChange={[Function]}
    removeResponse={[Function]}
    targetId="0"
    value={
      Object {
        "inTable": false,
      }
    }
  >
    A
  </DragSource(DropTarget(WithStyles(Component)))>
</span>
`;
